### PR TITLE
Pixelation widget now saves the width and height and bits per pixel

### PR DESCRIPTION
**Pull request by jmkulwik**
Prior to this change, the width and height of a student's project was not saved with the binary code. This meant that the image would display as a 10x10 image on page reload for projects that used V1 of the pixelation project. This was especially problematic for teachers who would load their student's images and not be able to see what the image was that the student created.

Note: I have changed the data schema of pixelation projects. Previously, V2 & V3 used the first two or three bytes, respectively, of the student code to define the width, height, and sometimes bitsPerPixel of the widget. Please take a close look and let me know if you foresee any potential issues.

Old Schema (In V2 & V3, the first two or three bytes were the width, height, and bitsPerPixel)
`Data: "101010..."`

New Schema
```
Data: {
  width: 12,
  height: 9,
  bitsPerPixel: 6,
  code: "101010..."
}
```
There were two other options I see for storing the width and height:
0. Migrate only V1 to the new model and migrate V2 & V3 later. 
1. Store it as two bytes at the beginning of the code string (what V2 & V3 do currently). Then, (to support old projects) add a flag to the project indicating that the first two bytes are the width & height
2. Migrate V2 & V3 to the new model as well. On serialize/de-serialize, add & remove those bytes to/from the code.

I don't feel too strongly about the approach I took, but these are the pros/cons I can see for the various options

0. Migrate only V1
  -- Pro: Simple, starts migrating us to a more resilient data model
  -- Con: We're now managing two different data models
1. Prepend the code string (what V2 & V3 did in the old model)
  -- Pro: Everything follows the current model
  -- Con: We'd have to do something odd to store the flag that indicates whether this is a legacy or "normal" version. Perhaps overload the getProjectHtml method to store the flag.
2. (New Model) Migrate V2 & V3 to the new object model as well.
  -- Pro: Everything is on the same model and is migrated to the more resilient data model
  -- Con: not much of a con. Just a bigger PR and a more projects that are on the legacy model.

## Links

- [jira](https://codedotorg.atlassian.net/browse/STAR-160)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/public/pixelation/pixelation.js [hunk 1]
```diff
@@ -154,14 +154,33 @@ function initProjects() {
...
               var binCode = pixel_data.value.replace(/[^01]/gi, "");
-              callback(isHexLevel() ? binToHexPvt(binCode) : binCode);
+              studentCode = isHexLevel() ? binToHexPvt(binCode) : binCode;
             }
+
+            var charactersToTrim = 0;
+            if (options.version === "2") {
+              charactersToTrim = 2;
+            } else if (options.version === "3") {
+              charactersToTrim = 3;
```
Comment (ajpal) [inline dashboard/public/pixelation/pixelation.js:173]: What's the 3rd character?

File: dashboard/public/pixelation/pixelation.js [hunk 2]
```diff
@@ -604,20 +662,30 @@ function getPositiveValue(element) {
...
+ */
+function getSliderBytes() {
   var heightByte = pad(getPositiveValue(heightRange).toString(2), 8, "0");
   var widthByte = pad(getPositiveValue(widthRange).toString(2), 8, "0");
   var bppByte = pad(getPositiveValue(bitsPerPixelRange).toString(2), 8, "0");
 
-  var justBits = pixel_data.value.replace(/[ \n]/g, "");
+  var sliderBits = widthByte + heightByte;
+  if (options.version === "3") {
+    sliderBits += bppByte;
```
Comment (ajpal) [inline dashboard/public/pixelation/pixelation.js]: what's bpp? Can this have a more descriptive name?

File: dashboard/public/pixelation/pixelation.js (hunk 1 above)
Comment (jmkulwik) [reply dashboard/public/pixelation/pixelation.js:173]: Fixed!

File: dashboard/public/pixelation/pixelation.js (hunk 2 above)
Comment (jmkulwik) [reply dashboard/public/pixelation/pixelation.js]: Fixed!

**Review by ajpal — APPROVED** [review]

**Pull request merged**